### DISCUSSION TITLE
How to hook operators?

### DISCUSSION THREAD

**Discussion by afk-Legacy** [vfsfitvnm/frida-il2cpp-bridge#116] [Q&A]
Hey! I wanted to ask regarding how can operators be hooked on, an example:

![image](https://user-images.githubusercontent.com/83068093/133547461-a3224986-7dba-427c-8227-8859c379e8bd.png)

**Comment by afk-Legacy** [comment]
This worked.

```
   EgoPoint.methods["op_Subtraction"].implementation = function(org : Il2Cpp.Object, value: Il2Cpp.Object) : Il2Cpp.Object
    {

        return this.methods["op_Addition"].invoke<Il2Cpp.Object>(org, value);
    }
```

**Comment by vfsfitvnm** [comment]
You have to follow [this](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/operator-overloads) table so you will know how these methods are named.

By the way, I recommend inspecting the dump produced by this package!

**Comment by afk-Legacy** [comment]
@vfsfitvnm Thanks, will definitely check it out. Btw, do you happen to have any snippets regarding Lists? I am trying to log a List here but facing errors.

**Comment by vfsfitvnm** [comment]
Do you mean Lists or Arrays? Because lists are just `Il2Cpp.Object` like anything else, arrays have their `Il2Cpp.Array` class. By the way, please open a new issue (if you think there is an issue) or a discussion at https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/categories/q-a
